website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -872,9 +872,8 @@ spec:
 A boolean field that controls whether instances created by Karpenter for this EC2NodeClass will have an associated public IP address. This overrides the `MapPublicIpOnLaunch` setting applied to the subnet the node is launched in. If this field is not set, the `MapPublicIpOnLaunch` field will be respected.
 
 {{% alert title="Note" color="warning" %}}
-If a `NodeClaim` requests `vpc.amazonaws.com/efa` resources, the `associatePublicIPAddress` field is ignored.
-A public IP address may only be associated with a node at launch if a single network interface is configured.
-This is inherently incompatible with instances configured for EFA workloads since Karpenter will configure an EFA for each network card on the instance.
+If a `NodeClaim` requests `vpc.amazonaws.com/efa` resources, `spec.associatePublicIPAddress` is respected. However, if this `NodeClaim` requests **multiple** EFA resources and the value for `spec.associatePublicIPAddress` is true, the instance will fail to launch. This is due to an EC2 restriction which
+requires that the field is only set to true when configuring an instance with a single ENI at launch. When using this field, it is advised that users segregate their EFA workload to use a separate `NodePool` / `EC2NodeClass` pair.
```

Comment:
Should we include the reference for customers?  https://docs.aws.amazon.com/cli/latest/reference/ec2/request-spot-fleet.html

Reply:
Looking at this ref again it's actually incorrect. Will probably drop it from the code, it describes the right behavior but for a different API. I've not been able to find any reference of this behavior for the APIs we use. I think it's fine for users to take our word for it, with this change they can try it themselves if they don't believe us 🤷. I would want to link if I could find a reference that actually applies to the APIs we use directly though.